internal/core/streams/stream_provider_server_side_flags.go

```diff
@@ -91,16 +97,44 @@ func (r *serverSideFlagsOnlyEnvStreamRepository) Replay(channel, id string) chan
 	}
 	go func() {
 		defer close(out)
-		if r.store.IsInitialized() {
-			flags, err := r.store.GetAll(ldstoreimpl.Features())
-
-			if err != nil {
-				r.loggers.Errorf("Error getting all flags: %s\n", err.Error())
-			} else {
-				out <- MakeServerSideFlagsOnlyPutEvent(
-					[]ldstoretypes.Collection{{Kind: ldstoreimpl.Features(), Items: removeDeleted(flags)}})
-			}
+		if !r.store.IsInitialized() {
+			return
+		}
+		flags, err := r.store.GetAll(ldstoreimpl.Features())
+
+		if err != nil {
+			r.loggers.Errorf("Error getting all flags: %s\n", err.Error())
+			return
+		}
+
+		checksum := hashKeyedItemDescriptors(flags)
+		r.mu.Lock()
+		defer r.mu.Unlock()
+		if r.flagsChecksum == checksum {
+			out <- r.cacheEvent
+			return
 		}
+
+		event := MakeServerSideFlagsOnlyPutEvent(
+			[]ldstoretypes.Collection{{Kind: ldstoreimpl.Features(), Items: removeDeleted(flags)}})
+		r.flagsChecksum = checksum
+		r.cacheEvent = event
+
+		out <- event
 	}()
 	return out
 }
+
+func hashKeyedItemDescriptors(flags []ldstoretypes.KeyedItemDescriptor) string {
+	h := sha256.New()
+	for _, flag := range flags {
```

Comment:
You'll want to sort these by key first for determinacy, because there's no guarantee that the items always come out of the data store in the same order.